Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -911,35 +911,37 @@ Binary analysis frameworks give you powerful ways to automate tasks that would b
 
-The target crackme is a simple license key validation Android app. Granted, you won't usually find license key validators like this, but the example should demonstrate the basics of static/symbolic analysis of native code. You can use the same techniques on Android apps that ship with obfuscated native libraries (in fact, obfuscated code is often put into native libraries specifically to make de-obfuscation more difficult).
+The target crackme is a simple license key validation Android executable. As we will soon observe, the key validation logic in the crackme is implemented in native code. It is a common notion that analyzing compiled native code is tougher than analyzing an equivalent compiled Java code, and hence, often business critical logic is written in native. The current sample application may not represent a real world problem, but nevertheless it provides us enough opportunities to learn the basics of symbolic execution to use them in a real situation. The techniques learned in this section will come in handy while dealing with obfuscated native code.
 
-The crackme takes the form of a native ELF binary that you can download here:
-
-<https://github.com/angr/angr-doc/tree/master/examples/android_arm_license_validation>
-
-Running the executable on any Android device should give you the following output:
+The crackme consists of a single ELF executable file, which can be executed on any Android device. To execute the binary file follow the instructions below:
 
 ```shell
 $ adb push validate /data/local/tmp
 [100%] /data/local/tmp/validate
+
 $ adb shell chmod 755 /data/local/tmp/validate
+
 $ adb shell /data/local/tmp/validate
 Usage: ./validate <serial>
+
 $ adb shell /data/local/tmp/validate 12345
 Incorrect serial (wrong format).
+
 ```
 
-So far so good, but you know nothing about what a valid license key looks like. Where do we start? Fire up Cutter to get a good look at what is happening. The main function is located at address 0x00001874 in the disassembly (note that this is a PIE-enabled binary, and Cutter chooses 0x0 as the image base address).
+So far so good, but we know nothing about what a valid license key looks like. To get started, open the ELF executable in a disassembler, like Cutter. The main function is located at offset `0x00001874` in the disassembly. It is important to note that this binary is PIE-enabled, and Cutter chooses to load the binary at `0x0` as image base address.
```

Comment:
I don't like "It is important to note that ...". This doesn't say much. Maybe explain why it's important, or just omit it

> Since this binary is PIE-enabled, Cutter can load the binary at `0x0` as image base address.